kernel/src/scheduler/cooperative.rs

```diff
@@ -76,8 +75,8 @@ impl<'a, C: Chip> Scheduler<C> for CooperativeSched<'a> {
             match node.proc {
                 Some(proc) => {
                     if proc.ready() {
-                        next = Some(proc.processid());
-                        break;
+                        let next = proc.processid();
+                        return SchedulingDecision::RunProcess((next, None));
                     }
                     self.processes.push_tail(self.processes.pop_head().unwrap());
```

Comment:
I think that we still want to execute this statement when iterating over a process, even if it'll be scheduled immediately.

Reply:
Are you talking about line 81? The original code exited the loop immediatelly when a suitable process was found. Furthermore, the [comments](https://github.com/tock/tock/blob/0a8847923a09dce9c2b7b03e4b7f81fc5e4f2f85/kernel/src/scheduler/cooperative.rs#L62) state that only empty processes slots or not-ready processes are placed at the back of the queue.

Do you still want me to change the cooperative scheduler implementation?

Reply:
Oh, I missed the `break`, sorry!